variables.tf

```diff
@@ -32,94 +44,105 @@ variable "allowed_ssh_ranges" {
-  type        = bool
-  default     = false
-  description = "Provision a static ip for use with managed zones/ingress"
+  default = {}
 }
 
-variable "google_dns_managed_zone" {
+variable "managed_dns" {
+  description = <<EOF
+  managed_dns = {
+    enabled = Whether to create DNS records in the given zone
+    name = Managed zone to modify
+    dns_name = DNS record names to create
+    service_prefixes = List of additional prefixes to the dns_name to create
+  }
+  EOF
   type = object({
-    enabled          = bool
-    name             = string
-    dns_name         = string
-    service_prefixes = set(string)
+    enabled          = optional(bool, false)
+    name             = optional(string, "")
+    dns_name         = optional(string, "")
+    service_prefixes = optional(set(string), [])
+
   })
-  default = {
-    enabled          = false
-    name             = ""
-    dns_name         = ""
-    service_prefixes = []
-  }
-  description = "Cloud DNS zone"
+  default = {}
 }
 
-variable "enable_advanced_datapath" {
-  description = "Enable the ADVANCED_DATAPATH provider"
-  type        = bool
-  default     = true
-}
+variable "kms" {
+  description = <<EOF
+  kms = {
+    database_encryption_key_name = Use an existing KMS key for the Application-layer Secrets Encryption settings. (Optional)
+  }
+  EOF
 
-variable "enable_network_policy" {
-  description = "Enable network policy switch. Has no effect when enable_advanced_datapath is true"
-  type        = bool
-  default     = true
-}
+  type = object({
+    database_encryption_key_name = optional(string, null)
+  })
 
-variable "enable_vertical_pod_autoscaling" {
-  type        = bool
-  default     = true
-  description = "Enable GKE vertical scaling"
+  default = {}
 }
 
-variable "gke_release_channel" {
-  type        = string
-  default     = "STABLE"
-  description = "GKE K8s release channel for master"
-}
+variable "gke" {
+  description = <<EOF
+  gke = {
+    k8s_version = Cluster k8s version
+    release_channel = GKE release channel
+    public_access = {
+      enabled = Enable API public endpoint
+      cidrs = List of CIDR ranges permitted for accessing the public endpoint
+    }
+    control_plane_ports =  Firewall ports to open from the master, e.g., webhooks
+    advanced_datapath = Enable the ADVANCED_DATAPATH provider
+    network_policies = Enable network policy switch. Has no effect when enable_advanced_datapath is true
+    vertical_pod_autoscaling = Enable GKE vertical scaling
+    kubeconfig = {
+      path = Specify where the cluster kubeconfig file should be generated.
+    }
+  }
+  EOF
 
-variable "location" {
-  type        = string
-  default     = "us-west1-b"
-  description = "The location (region or zone) of the cluster. A zone creates a single master. Specifying a region creates replicated masters accross all zones"
-}
+  type = object({
+    k8s_version     = optional(string, ""),
+    release_channel = optional(string, "RAPID"),
```

Comment:
so we are changing the default? my only concern is that we would end up with 1.29 before we expected to support it.

Reply:
yeah, I thought this aligned better with what we're using internally. should I maybe set `k8s_version`?

Reply:
well, you need both. the channel restricts the version.